### ISSUE TITLE
invalid argument 0: hex number with leading zero digits

### ISSUE THREAD

**Issue by WyseNynja** [0xProject/0x-mesh#305]
I'm seeing this a lot in my logs. At first I thought it was because my node wasn't synced, but it finished syncing this afternoon so I started 0x-mesh back up.

```
zrx_mesh_1         | {"error_rpc_jsonError":"invalid argument 0: hex number with leading zero digits","level":"error","msg":"BlockWatcher error encountered","myPeerID":"16Uiu2HAm8o1zLU3mkpC9aZXmFajoQxkkfL9HQhwCQahLfXS5hMxT","time":"2019-07-25T03:01:16Z"}
```

The geth logs show a little different information:

```
INFO [07-25|03:01:10.182] Imported new block headers               count=1    elapsed=7.004ms    number=8217268 hash=fe753b…7ada06
WARN [07-25|03:01:11.887] Served eth_getBlockByNumber              conn=127.0.0.1:37294 reqid=447 t=33.326µs  err="invalid argument 0: hex number with leading zero digits"
INFO [07-25|03:01:13.347] Imported new block headers               count=1    elapsed=6.858ms    number=8217269 hash=f3e552…2dddf2
WARN [07-25|03:01:16.887] Served eth_getBlockByNumber              conn=127.0.0.1:37294 reqid=448 t=28.233µs  err="invalid argument 0: hex number with leading zero digits"
WARN [07-25|03:01:21.887] Served eth_getBlockByNumber              conn=127.0.0.1:37294 reqid=449 t=40.979µs  err="invalid argument 0: hex number with leading zero digits"
```

I do see lots of `"msg":"received new valid order from peer"`, too. So they aren't all failing.

**Comment by WyseNynja** [comment]
I'm also seeing these errors:
```
zrx_mesh_1         | {"attempt_number":0,"error_string":"no contract code at given address","level":"info","msg":"GetOrderRelevantStates request failed","myPeerID":"16Uiu2HAm8o1zLU3mkpC9aZXmFajoQxkkfL9HQhwCQahLfXS5hMxT","numOrders_number":134,"time":"2019-07-25T03:08:35Z"}
zrx_mesh_1         | {"attempt_number":1,"error_string":"no contract code at given address","level":"info","msg":"GetOrderRelevantStates request failed","myPeerID":"16Uiu2HAm8o1zLU3mkpC9aZXmFajoQxkkfL9HQhwCQahLfXS5hMxT","numOrders_number":134,"time":"2019-07-25T03:08:35Z"}
zrx_mesh_1         | {"attempt_number":2,"error_string":"no contract code at given address","level":"info","msg":"GetOrderRelevantStates request failed","myPeerID":"16Uiu2HAm8o1zLU3mkpC9aZXmFajoQxkkfL9HQhwCQahLfXS5hMxT","numOrders_number":134,"time":"2019-07-25T03:08:35Z"}
zrx_mesh_1         | {"attempt_number":3,"error_string":"no contract code at given address","level":"info","msg":"GetOrderRelevantStates request failed","myPeerID":"16Uiu2HAm8o1zLU3mkpC9aZXmFajoQxkkfL9HQhwCQahLfXS5hMxT","numOrders_number":134,"time":"2019-07-25T03:08:36Z"}
```

Should I open this as a separate issue, or do you think it is related?

**Comment by albrow** [comment]
@WyseNynja what version of Mesh are you running? (Mesh prints out the version on initialization but you can also look in the source code).

Can you provide more information about what steps led up to this error?

**Comment by WyseNynja** [comment]
I'm using 0xorg/mesh:latest, which is currently at 1.0.6-beta.

I've seen similar errors in the past with other projects when my node wasn't synced. I just checked now and my node is behind again. I just realized that it is on my ZFS raid array and not my SSD.  I guess ZFS's RAM caching isn't enough for geth to keep up. I'll move it to the SSD next I have time and see if that helps.

**Comment by albrow** [comment]
Thanks @WyseNynja and let me know what you find. We are currently monitoring errors and warnings across all nodes which have enabled telemetry and I'm not seeing this error anywhere else.

Can you share more about your configuration? What environment variables are you using?

**Comment by albrow** [comment]
I think this was fixed in #353. @WyseNynja would you mind confirming? The latest version as of now is `v3.0.1-beta`.

**Comment by WyseNynja** [comment]
I'm not seeing these errors anymore with the latest version. Thank you!
